app/datasources/db/models.py

```diff
@@ -78,3 +94,17 @@ class Contract(SqlQueryBase, SQLModel, table=True):
     )
     project: Project | None = Relationship(back_populates="contracts")
     chain_id: int = Field(default=None)
+
+    @classmethod
+    async def get_abi_by_contract_address(
+        cls, session: AsyncSession, address: bytes
+    ) -> ABI | None:
+        results = await session.exec(
+            select(Abi.abi_json)
+            .join(cls)
+            .where(cls.address == address)
+            .where(cls.abi_id == Abi.abi_hash)
+        )
+        if result := results.first():
```

Comment:
In this case we will return an ABI for any chain, the ABI could be the not expected, could you add a comment with a TODO to add support for multichain?

Reply:
Sure